src/main/java/fr/xephi/authme/service/GeoIpService.java

```diff
@@ -1,46 +1,80 @@
 package fr.xephi.authme.service;
 
 import com.google.common.annotations.VisibleForTesting;
-import com.maxmind.geoip.LookupService;
+import com.google.common.hash.HashCode;
+import com.google.common.hash.HashFunction;
+import com.google.common.hash.Hashing;
+import com.google.common.io.Resources;
+import com.ice.tar.TarEntry;
+import com.ice.tar.TarInputStream;
+import com.maxmind.db.GeoIp2Provider;
+import com.maxmind.db.Reader;
+import com.maxmind.db.Reader.FileMode;
+import com.maxmind.db.cache.CHMCache;
+import com.maxmind.db.model.Country;
+import com.maxmind.db.model.CountryResponse;
+
 import fr.xephi.authme.ConsoleLogger;
 import fr.xephi.authme.initialization.DataFolder;
-import fr.xephi.authme.util.FileUtils;
 import fr.xephi.authme.util.InternetProtocolUtils;
 
-import javax.inject.Inject;
+import java.io.BufferedInputStream;
 import java.io.File;
-import java.io.FileOutputStream;
 import java.io.IOException;
-import java.io.InputStream;
 import java.io.OutputStream;
+import java.net.InetAddress;
 import java.net.URL;
-import java.net.URLConnection;
-import java.util.concurrent.TimeUnit;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.FileTime;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Objects;
+import java.util.Optional;
 import java.util.zip.GZIPInputStream;
 
-import static com.maxmind.geoip.LookupService.GEOIP_MEMORY_CACHE;
+import javax.inject.Inject;
 
 public class GeoIpService {
+
     private static final String LICENSE =
-        "[LICENSE] This product uses data from the GeoLite API created by MaxMind, available at http://www.maxmind.com";
-    private static final String GEOIP_URL =
-        "http://geolite.maxmind.com/download/geoip/database/GeoLiteCountry/GeoIP.dat.gz";
-    private LookupService lookupService;
-    private Thread downloadTask;
+            "[LICENSE] This product includes GeoLite2 data created by MaxMind, available at https://www.maxmind.com";
+
+    private static final String DATABASE_NAME = "GeoLite2-Country";
+    private static final String DATABASE_EXT = ".mmdb";
+    private static final String DATABASE_FILE = DATABASE_NAME + DATABASE_EXT;
+
+    private static final String ARCHIVE_FILE = DATABASE_NAME + ".tar.gz";
+
+    private static final String ARCHIVE_URL = "https://geolite.maxmind.com/download/geoip/database/" + ARCHIVE_FILE;
+    private static final String CHECKSUM_URL = ARCHIVE_URL + ".md5";
 
-    private final File dataFile;
+    private static final int UPDATE_INTERVAL = 30;
```

Comment:
Maybe here add _DAYS to the name to make the unit clear—I had no idea until I saw its usage.

Reply:
You're right. I'm going to change that.